stdlib/public/runtime/Casting.cpp

```diff
@@ -949,7 +966,7 @@ static bool _dynamicCastToExistential(OpaqueValue *dest,
     auto object = *(reinterpret_cast<HeapObject**>(srcDynamicValue));
     destExistential->Value = object;
     if (!canConsumeDynamicValue || !(flags & DynamicCastFlags::TakeOnSuccess)) {
-      swift_retain(object);
+      swift_unknownRetain(object);
```

Comment:
Because object is not necessarily a native swift object. (I ran into this in a test)

We probably did never ran into this before because before my change canConsumeDynamicValue is true and address casts probably always 'take'.